### ISSUE TITLE
CI - mspec intermittent failures

### ISSUE THREAD

**Issue by MSP-Greg** [oneclick/rubyinstaller2#156]
Saw the somewhat intermittent mspec failures.  I have no idea what's causing them, but, JFYI:

1. ruby/ruby still runs mspec without the `-j` argument.  They do run the spec suite on Ruby 2.5 and later, but if `test-all` fails, the job stops, so...

2. Both ruby/spec and ruby-loco have `-rdevkit` as an argument.

3. It's been a long time since I've had to change the base test code in ruby-loco, but I don't think I would have added the `-rdevkit` argument without a reason.

Then again, it might still have the intermittent issues with the above...

Thanks, and hope things are well, Greg

**Comment by larskanis** [comment]
Hi @MSP-Greg ! Are you speaking about [these segfaults](https://ci.appveyor.com/project/larskanis/rubyinstaller2-hbuor/branch/master/job/xq0wylggjr8hyevo#L2955)? Or something different?

I already investigated the segfault issue a bit deeper, but couldn't find the root issue. It has nothing to do with the mspec `-j` flag, with `-rdevkit` and is not caused by a single test. But I was able to separate and simplify the issue into two single tests, which when executed together causes the ruby core to segfault. It seems to be caused by some particular memory fragmentation so that the ruby VM crashes while some binding operations. It disappeared by running gcc with `-O3` argument through [this commit](https://github.com/oneclick/rubyinstaller2-packages/commit/4b92a84e8a09d948402f2c4deff3ff1f1d21c702), which I used for the last series of rubyinstaller releases.

Now the segfaults reappeared in ruby-2.5.6-x64 and they are currently the blocker for rubyinstaller-2.5.6. Do you have the segfaults with ruby-loco as well?

**Comment by MSP-Greg** [comment]
Lars,

Good day.

Re the tests, builds 1137 thru 1140, which all fail on 2.5.6-x64 & 2.7.0-x86, but each consistently fails on a different spec test.  Didn't notice that before.  I've just looked at a sample of the builds.

> Do you have the segfaults with ruby-loco as well?

No.  But, I haven't built anything but trunk 64 for a long time.  When ruby/ruby added full testing for MinGW, they didn't want to use makepkg, so, along with helping with that, I removed it from ruby-loco.  Since then, I haven't even set up my code to switch between 64 & 32.

>  I was able to separate and simplify the issue into two single tests, which when executed together causes the ruby core to segfault.

Thanks for tracking that down.  Sounds non-trivial to find.  Parallel testing is a PITA sometimes.

If there's anyway I can help, let me know.  I may try getting ruby-loco to build 32 bit.

Tough call, ruby/ruby testing (non-parallel) passes, but parallel doesn't.  Release or not?

Lastly, we're running spec parallel, maybe it's time to see if ruby/spec and ruby/ruby can do the same...
